File: database/database-annotation-processor/src/test/java/ru/tinkoff/kora/database/common/annotation/processor/r2dbc/R2dbcEntity.java
package ru.tinkoff.kora.database.common.annotation.processor.r2dbc;

import io.r2dbc.spi.Row;
import io.r2dbc.spi.Statement;
import ru.tinkoff.kora.database.common.annotation.processor.entity.TestEntityRecord;
import ru.tinkoff.kora.database.r2dbc.mapper.parameter.R2dbcParameterColumnMapper;
import ru.tinkoff.kora.database.r2dbc.mapper.result.R2dbcResultColumnMapper;
import ru.tinkoff.kora.database.r2dbc.mapper.result.R2dbcRowMapper;

import javax.annotation.Nullable;
import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;

public class R2dbcEntity {
    public record AllNativeTypesEntity(
        boolean booleanPrimitive,
        @Nullable Boolean booleanBoxed,
        int integerPrimitive,
        @Nullable Integer integerBoxed,
        long longPrimitive,
        @Nullable Long longBoxed,
        double doublePrimitive,
        @Nullable Double doubleBoxed,
        String string,
        BigDecimal bigDecimal,
        byte[] byteArray,
        LocalDateTime localDateTime,
        LocalDate localDate
    ) {}

    public static final class TestEntityFieldR2dbcResultColumnMapper implements R2dbcResultColumnMapper<TestEntityRecord.MappedField1> {
        @Override
        public TestEntityRecord.MappedField1 apply(Row row, String label) {
            return null;
        }
    }

    public static class TestEntityFieldR2dbcResultColumnMapperNonFinal implements R2dbcResultColumnMapper<TestEntityRecord.MappedField2> {
        @Override
        public TestEntityRecord.MappedField2 apply(Row row, String label) {
            return null;
        }
    }

    public static final class TestEntityR2dbcRowMapper implements R2dbcRowMapper<TestEntityRecord> {
        @Override
        public TestEntityRecord apply(Row row) {
            return null;
        }
    }

    public static class TestEntityR2dbcRowMapperNonFinal implements R2dbcRowMapper<TestEntityRecord> {
        @Override
        public TestEntityRecord apply(Row row) {
            return null;
        }
    }

    public static final class TestEntityFieldR2dbcParameterColumnMapper implements R2dbcParameterColumnMapper<TestEntityRecord.MappedField1> {
        @Override
        public void apply(Statement stmt, int index, TestEntityRecord.MappedField1 value) {

        }
    }

    public static class TestEntityFieldR2dbcParameterColumnMapperNonFinal implements R2dbcParameterColumnMapper<TestEntityRecord.MappedField2> {
        @Override
        public void apply(Statement stmt, int index, TestEntityRecord.MappedField2 value) {

        }
    }
}

File: database/database-symbol-processor/src/test/kotlin/ru/tinkoff/kora/database/symbol/processor/r2dbc/R2dbcEntity.kt
package ru.tinkoff.kora.database.symbol.processor.r2dbc

import io.r2dbc.spi.Row
import io.r2dbc.spi.Statement
import ru.tinkoff.kora.database.r2dbc.mapper.parameter.R2dbcParameterColumnMapper
import ru.tinkoff.kora.database.r2dbc.mapper.result.R2dbcResultColumnMapper
import ru.tinkoff.kora.database.r2dbc.mapper.result.R2dbcRowMapper
import ru.tinkoff.kora.database.symbol.processor.entity.TestEntity
import java.math.BigDecimal
import java.time.LocalDate
import java.time.LocalDateTime

data class AllNativeTypesEntity(
    val booleanPrimitive: Boolean,
    val booleanBoxed: Boolean?,
    val integerPrimitive: Int,
    val integerBoxed: Int?,
    val longPrimitive: Long,
    val longBoxed: Long?,
    val doublePrimitive: Double,
    val doubleBoxed: Double?,
    val string: String?,
    val bigDecimal: BigDecimal?,
    val byteArray: ByteArray?,
    val localDateTime: LocalDateTime?,
    val localDate: LocalDate?
)


class TestEntityFieldR2dbcResultColumnMapper : R2dbcResultColumnMapper<TestEntity.MappedField1> {
    override fun apply(row: Row, label: String): TestEntity.MappedField1 {
        return TestEntity.MappedField1()
    }
}

open class TestEntityFieldR2dbcResultColumnMapperNonFinal : R2dbcResultColumnMapper<TestEntity.MappedField2> {
    override fun apply(row: Row, label: String): TestEntity.MappedField2 {
        return TestEntity.MappedField2()
    }
}

class TestEntityR2dbcRowMapper : R2dbcRowMapper<TestEntity> {
    override fun apply(row: Row): TestEntity {
        return null!!
    }
}

open class TestEntityR2dbcRowMapperNonFinal : R2dbcRowMapper<TestEntity> {
    override fun apply(row: Row): TestEntity {
        return null!!
    }
}

class TestEntityFieldR2dbcParameterColumnMapper : R2dbcParameterColumnMapper<TestEntity.MappedField1?> {
    override fun apply(stmt: Statement, index: Int, o: TestEntity.MappedField1?) {}
}

open class TestEntityFieldR2dbcParameterColumnMapperNonFinal : R2dbcParameterColumnMapper<TestEntity.MappedField2?> {
    override fun apply(stmt: Statement, index: Int, o: TestEntity.MappedField2?) {}
}
